### ISSUE TITLE
regular accessioning activity can crash common-accessioning VMs as currently provisioned

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/common-accessioning#1069]
**Describe the bug**
We've seen common-accessioning prod VMs lock up and become unresponsive at least a couple times in the last few weeks, seemingly due to load being too high.  A reboot of the locked up VMs was needed to get them working again.

We suspect (based on circumstantial evidence) that jp2 creation is the main culprit for this overload, as it seems like jp2 creation was very active at the times when the VMs crashed, and we know it's a very resource intensive activity.  e.g. @andrewjbtw [observed](https://stanfordlib.slack.com/archives/C09M7P91R/p1687542297634019) on fri 2023-06-23: "A large batch of images just came in in a short period and now about 80 of them are in error states at jp2-create, more evidence that the issue is related to jp2-create. The druids in this batch only have a small number of images in them - looks like front and back of photographs - but they all look in the 100-200 MB range."

@rsmith11 also rebooted common-accessioning after a similar crash the day before, thu 2023-06-22, [see here](https://stanfordlib.slack.com/archives/C09M7P91R/p1687470894362989).

**User Impact**
Regular accessioning activity is slowed down, both because some work-in-progress crashes and gets stuck, and because less resourcing is available for other accessioning jobs as available VMs get overloaded (with some possibly crashing and becoming unavailable).

Additionally, jobs that get picked up and which hang and crash a VM require someone (usually @andrewjbtw) to watch the Argo workflow grid, and either 1) retry things in an error state,  or worse, 2) reset the step manually from the [workflow service](https://github.com/sul-dlss/workflow-server-rails/) Rails console if the common-accessioning job didn't crash in a way that resulted in an error being recorded for the workflow step (because the reboot happened while the workflow step was in the "Started" state, and the Sidekiq processes were brought down without raising an error that the robot framework could catch and record).

TLDR -- accessioning slowdown and instability, time sinks for infra developers and the repository manager.

**To Reproduce**

let common-accessioning run in prod for a week or two

**Expected behavior**

common-accessioning VMs don't crash due to regularly expected load

**Screenshots**

n/a

**Additional context / TODO**

We doubled the number of jp2 create processes because work was piling up on the jp2 create step.  Since it seems like this number of workers would keep up with the expected accessioning volume if the VMs don't crash, we determined in post-standup discussion that the easy-ish fix here is:

- [x] file an ops ticket to double the number of common-accessioning VMs -- https://github.com/sul-dlss/operations-tasks/issues/3403
- [x] wait for new VMs to be created
- [x] update sidekiq process related configs (in shared_configs and puppet), and cap deployment configs (only for prod)
  - [x] PR to `config/deploy/prod.rb` to list the 4 new VMs
  - [x] PRs to halve sidekiq worker allocation on each VM
    - [x] common-accessioning shared_configs PR to halve the number of Sidekiq workers on any given VM.  Currently, the Sidekiq configs allocate two worker management processes for all of the non-`assemblyWF_jp2` queues, and two worker management processes solely for the `assemblyWF_jp2` queue.  To stay closest to the current process/thread allocation...  We'll halve the concurrency for each of the non-`assemblyWF_jp2` handling processes (each of which has 4 threads allocated).  But we can't do that for the `assemblyWF_jp2` worker mgmt processes (each of which only have 1 thread allocated at present), so we'll just drop one `assemblyWF_jp2` process.
      - [x] puppet PR to adjust the expected sidekiq worker manager process count on each common-accessioning prod VM.
- [x] get PRs reviewed/approved
  - [x] https://github.com/sul-dlss/common-accessioning/pull/1072
  - [x] https://github.com/sul-dlss/shared_configs/pull/2014
  - [x] https://github.com/sul-dlss/puppet/pull/9679
- [x] merge them all at (around) the same time.  if they don't all get merged at the same time, i think the safest order would be the puppet PR, then the shared_configs PR, then the common-accessioning PR.  that way, we're decreasing the common-accessioning worker count before increasing the number of common-accessioning VMs that are online.  this might lead to a brief backlog of jobs, but seems better than having too many workers online for a bit and possibly DoSing our other services?
- [x] get the puppet changes applied on all 8 VMs
- [x] per https://github.com/sul-dlss/puppet/pull/9679#issuecomment-1633190655 make sure that the sidekiq `sidekiq-4.service` file that the puppet PR intends to do away with is gone (and remove manually if it isn't)
- [x] deploy to prod (possibly with only one of the new boxes in the first deployment, in case we missed something and run into snags, then we only have one box to debug or take back offline, instead of 4 new ones)

**Comment by jmartin-sul** [comment]
@rsmith11 says the new common-accessioning VMs are ready to go https://github.com/sul-dlss/operations-tasks/issues/3403#issuecomment-1615068164 --

> the 4 servers are built with the same configs and a-d. No app installed on them yet, but should be ready to go otherwise.
>
> Once they are in production, I'll add them to nagios production monitoring.

**Comment by jmartin-sul** [comment]
deployed to the new VMs and deployed the new configs to all the VMs.  all seems well and work is successfully getting picked up on the new machines.  closing as done.
